semanticdb/semanticdb3/semanticdb3.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package scala.meta.internal.semanticdb3;
+
+message TextDocuments {
+  repeated TextDocument documents = 1;
+}
+
+message TextDocument {
+  string format = 1;
+  string uri = 2;
+  string text = 3;
+  string language = 4;
+  repeated SymbolInformation symbols = 5;
+  repeated SymbolOccurrence occurrences = 6;
+  repeated Diagnostic diagnostics = 7;
+  repeated Synthetic synthetics = 8;
+}
+
+message Range {
+  int32 start_line = 1;
+  int32 start_character = 2;
+  int32 end_line = 3;
+  int32 end_character = 4;
+}
+
+message SymbolInformation {
+  enum Kind {
+    UNKNOWN1 = 0;
+    VALUE = 1;
+    VARIABLE = 2;
+    METHOD = 3;
+    PRIMARY_CONSTRUCTOR = 4;
+    SECONDARY_CONSTRUCTOR = 5;
+    MACRO = 6;
```

Comment:
DEF_MACRO? Maybe `macro class` will become a thing one day, who knows

Reply:
I suggest we keep MACRO. If more language features are introduced, we can add them later. Even renames aren't off the table imo, because: 1) SemanticDB is not a public API, 2) the protobuf schema doesn't care.

Reply:
fair enough